src/scripts.c

```diff
@@ -51,7 +51,11 @@ static void update_triggers(void)
 {
 	__attribute__((unused)) int ret = 0;
 
-	ret = system("/usr/bin/systemctl --no-block daemon-reload");
+	if (need_systemd_reexec) {
+		ret = system("/usr/bin/systemctl daemon-reexec");
+	}
```

Comment:
I may have misunderstood the intent of removing --no-block here. Tudor, can you clarify that?